client/server/pages/index.js

```diff
@@ -109,6 +112,24 @@ function setupLoggedInContext( req, res, next ) {
 	next();
 }
 
+async function setupDesktopUserAuth( req, res, next ) {
```

Comment:
I'm not very familiar with the boot logic and not sure if I'm on the right path here.

Problem: this doesn't actually work (app boots to a blank white screen). Also the conditional import of `electron/ipc` is laughable here, although that being said I'm not entirely sure how to insert the `user-auth` IPC event within the existing boot sequence.

Also, one interesting fact about the Electron side of things is that the main window is created _after_ the Calypso server has completed initializing ([link](https://github.com/Automattic/wp-calypso/blob/6e1f5f5ed209c216a8751ec28031199b7e6accf8/client/desktop/server/index.js#L151)). Cookies sent over IPC are parsed and saved to the web session of an already-created window object ([link](https://github.com/Automattic/wp-calypso/blob/6e1f5f5ed209c216a8751ec28031199b7e6accf8/client/desktop/lib/cookie-auth/index.js#L73-L96)). 

TL;DR If the Electron window is created _after_ the server finishes initializing, and if the window is required to parse and set cookies relayed by IPC, not sure if/how this general approach will work. 🤔